docs/en/rules/Azure.VM.MaintenanceConfig.md

```diff
@@ -0,0 +1,81 @@
+---
+severity: Important
+pillar: Operational Excellence
+category: Repeatable infrastructure
+resource: Virtual Machine
+online version: https://azure.github.io/PSRule.Rules.Azure/en/rules/Azure.VM.MaintenanceConfig/
+---
+
+# Associate a maintenance configuration
+
+## SYNOPSIS
+
+Use a maintenance configuration for virtual machines.
+
+## DESCRIPTION
+
+Virtual machines can be attached to a maintenance configuration which allows customer managed assessments and updates for machine patches within the guest operating system.
+
+## RECOMMENDATION
+
+Consider associating a maintenance configuration for virtual machines.
+
+## EXAMPLES
+
+### Configure with Azure template
+
+To deploy virtual machines that pass this rule:
+
+- Deploy a `Microsoft.Maintenance/configurationAssignments` sub-resource (extension resource).
+- Set the `properties.maintenanceConfigurationId` property to the linked maintenance configuration resource Id.
+
+For example:
+
+```json
+{
+  "type": "Microsoft.Maintenance/configurationAssignments",
+  "apiVersion": "2022-11-01-preview",
+  "name": "[parameters('assignmentName')]",
+  "location": "[parameters('location')]",
+  "scope": "[format('Microsoft.Compute/virtualMachines/{0}', parameters('name'))]",
+  "properties": {
+    "maintenanceConfigurationId": "[format('/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/test-rg/providers/Microsoft.Maintenance/maintenanceConfigurations/{0}', parameters('maintenanceConfigName'))]",
+  },
+  "dependsOn": [
+    "[resourceId('Microsoft.Compute/virtualMachines', parameters('name'))]"
+  ]
+}
+```
+
+### Configure with Bicep
+
+To deploy virtual machines that pass this rule:
+
+- Deploy a `Microsoft.Maintenance/configurationAssignments` sub-resource (extension resource).
+- Set the `properties.maintenanceConfigurationId` property to the linked maintenance configuration resource Id.
+
+For example:
+
+```bicep
+resource assosciateMaintenanceConfig 'Microsoft.Maintenance/configurationAssignments@2022-11-01-preview' = {
+  name: assignmentName
+  location: location
+  scope: vm
+  properties: {
+    maintenanceConfigurationId: maintenanceConfigurationId
+  }
+}
+```
+
+## NOTES
+
+Not all operating systems are supported, check out the `LINKS` section for more information. Update management center doesn't support driver updates.
```

Comment:
```suggestion
Operating system updates with Update Managment center is a preview feature.
Not all regions or operating systems are supported, check out the `LINKS` section for supported regions.
Update management center doesn't support driver updates.
```

Reply:
@BernieWhite Minor feedback from me, for Azure VMs all regions are supported, compared to Arc-enabled servers.

`Update management center (preview) is available in all Azure public regions where compute virtual machines are available.`

https://learn.microsoft.com/azure/update-center/support-matrix?tabs=azurevm%2Cazurevm-os#supported-regions

Should we add this anyways?

Reply:
Good point @BenjaminEngeset. We should still add the preview comment, but no comment about region limiting.

For example:

> Operating system updates with Update Managment center is a preview feature.
> Update management center doesn't support driver updates.

Reply:
@BernieWhite Check latest update from me now and see if you think this is suitable.

Reply:
@BenjaminEngeset Looks good to me.